### DISCUSSION TITLE
Can't Access Repository Description And It's Classifying Topics Using A Screen Reader

### DISCUSSION THREAD

**Discussion by faddymichel** [community/community#23984] [New to GitHub]
<p>Hi all,</p>
<p>I’m a completely blind person trying to edit the description of a repository, and classify it through topics. I have tried to follow the steps found on this <a href="https://docs.github.com/en/github/administering-a-repository/classifying-your-repository-with-topics#adding-topics-to-your-repository" rel="noopener nofollow ugc">GitHub Help Article</a>, but I keep failing on finding the “About” and the icon next to it. I have also tried following the same steps in various repositories, all owned by me, without better luck. I mainly use the ORCA screen reader, but the ChromeVox screen reader couldn’t help either.</p>
<p>Best,</p>
<p>Faddy Michel<br>
| Independent Software Engineer<br>
| Twitter: <a href="https://twitter.com/faddymichel" rel="noopener nofollow ugc">@faddymichel</a></p>

**Comment by faddymichel** [comment, marked as answer]
<h3>Update!</h3>
<p>I asked my friend who’s blessed with sharp vision (<a href="https://github.com/mark7389" rel="noopener nofollow ugc">@mark7389</a>) to check if he can see and access the “About” section on a repository main page; and edit it’s metadata information. Of course, since Accessibility is NOT a question for sighted users, he was able to see and access that section without issues.</p>
<h4>If You Are Sighted:</h4>
<p>You can edit the Repository Metadata Information, just fine, from any browser.</p>
<h4>But, If You Are Blind:</h4>
<p>You can only edit the Repository Metadata Information from Firefox.</p>
<h2>Dear GitHub Developers and Designers</h2>
<p>Please put the Blind in Mind while Designing and Developing any of your products. Because, honestly, you are ruining Git in this manner; Git is Fully Accessible and Usable for, pretty much, everyone. You can definitely learn from Git how to achieve Accessibility and Usability without any Dirty fixes.</p>

**Comment by canuckjacq** [comment]
<p>:wave: Welcome!</p>
<p>Thanks for bringing this to our attention. I flagged this for our accessibility folks yesterday and they tested it and said that there is an error in the focus order on the repository page: you’ll hear the “edit repository metadata” before the “about” heading, which is 1. incorrect order and 2. not the wording we use in our documentation. This was specifically tested in Edge in case that helps.</p>
<p>We’ve opened an issue to fix the documentation and the focus order. I’m sorry this was incorrect in the first place.</p>

**Comment by faddymichel** [comment]
<p>Hi canuckjacq,</p>
<p>Thanks for your response. The issue isn’t only about the order of focus and the wording of the documentation. The “about” and the “Edit repository metadata information” aren’t showing at all on Chromium and Google Chrome when using ORCA and Chromevox. I found them only showing on Firefox. After I asked my friend to inspect the code, we found that this button is created inside an SVG element. Honestly, I don’t see any necessity for SVG to be holding that button. They can use the <code>&lt;button&gt;</code> element and put the SVG inside of it. Also, they have to read more on Accessibility issues with SVG on the MDN docs.</p>
<h3>Suggestion</h3>
<p>It would be more helpful if the repository metadata information (description, topics, etc.) can be edited on the “Settings” page (right below the repository name).</p>
<p>Again, thanks and hope my response would help fixing the issue in a smoother way.</p>

**Comment by canuckjacq** [comment]
<p>Thank you so much for walking me through this. I know it’s exhausting to have to point this stuff out and explain things all the time. I have shared your explanations with our accessibility folks and they immediately recognised that this pattern goes against W3C recommendations, for exactly this reason.</p>
<p>I hope we’ll have this resolved for you and everyone else soon. Please never hesitate to bring accessibility barriers to our attention - we really do want <em>everyone</em> to be able to code and collaborate as easily as possible on GitHub.</p>

**Comment by faddymichel** [comment]
<p>Great! I’m glad that you guys are being so collaborative with me on that matter.</p>
<p>Something caught my attention when you said that you want everyone to be able to code and collaborate as easily as possible on GitHub. For this to happen, I need to not only <em><strong>access</strong></em> but to <em><strong>use</strong></em> as well. GitHub’s main and community sites are <em><strong>accessible</strong></em> (there might be minor bugs here and there). But when it comes to <em><strong>usability</strong></em>, excuse my honesty here, both sites are far from being <em><strong>usable</strong></em>. It’s pretty much like giving a tiny bike to a big person and ask them keep riding it all the time; they’d actually use it as briefly as possible if really needed. What I mean here is that GitHub products need to be designed from the beginning to include everyone (the blind is one segment). Obviously, this is not the case. What happens here is that the design is made from the beginning in a very visual and graphical manner and then see how to make this graphical interface accessible.</p>
<p>Anyways, I’ll keep doing my part and raise such bugs, and I won’t stop pointing out that GitHub <strong>must</strong> work in a different manner to really achieve <em><strong>usability</strong></em> someday.</p>
